src/main/java/org/pac4j/dropwizard/Pac4jBundle.java

```diff
@@ -43,8 +44,39 @@ public final void run(T configuration, Environment environment)
             }
         }
 
-        environment.jersey().register(new Pac4JSecurityFeature(config));
+        String urlPrefix = getUrlPrefix(configuration, environment);
+
+        environment.jersey()
+                .register(new Pac4JSecurityFeature(config, urlPrefix));
         environment.jersey()
                 .register(new Pac4JValueFactoryProvider.Binder(config));
     }
+
+    private String getUrlPrefix(T configuration, Environment environment) {
+        // the context path on the
+        String cPath = environment.getApplicationContext().getContextPath();
+        String sPath = ((AbstractServerFactory) configuration
+                .getServerFactory()).getJerseyRootPath().orElse("");
+
+        return (clean(cPath) + clean(sPath)).replace("//", "/");
+    }
+
+    private String clean(String path) {
```

Comment:
I know, I guess it can be improved, for example using existing libraries doing that better than I... For example, we must be careful not to leave two "/" in front of the prefix, if not it will be interpreted as an absolute url and when doing redirect, it won't work! It took me some time to actually write that, I'm pretty sure it is not optimal!